mozilla/bedrock
# 10650 + 10504: Contentful-powered VPN Resource Centre pages

## Description

This changeset takes the FE work done by @nathan-barrett  on #10504 and builds in backend changes that pull data from Contentful for the VRC and displays it via the Nathan's FE.

It's not small, but it's fairly well separated in terms of reviewing.

Notes on behaviour:
* update_contentful management command now stores the slug for the page
* page data from bedrock.contenful.api now also stores the SEO object from a Compose page (**only** if available), which, in this situation, we use in the card for the listing page
* the listing page defaults to showing articles listed by most recently updated/synced 
* use active_locales to steer users to en-US (the current only supported lang for now)
* raise a 404 when we really can't find a match for a given slug

Some tests have been added, and more are being written while this is in initial review. The balance will be a fast follow.

WIP screenshots:

![Screenshot 2021-11-04 at 10-52-10 Knowledge Center](https://user-images.githubusercontent.com/101457/140392881-de263be3-bb7d-4c9c-acec-607993792fad.png)

![Screenshot 2021-11-04 at 17-52-48 Four ways to protect your data privacy and still be online](https://user-images.githubusercontent.com/101457/140393079-633b4fb0-8611-4bc0-958a-7817aa471480.png)

## Issue / Bugzilla link

#10650 and #10504

Also resolves #10712 which has been merged into this branch

## Testing

* ensure you have Contentful API keys locally. Set your `CONTENTFUL_ENVIRONMENT` to `sandbox`.  
* `make shell`
* `./manage.py update_contentful --force`
* review what is logged - the syncing of `pageHome` as well as `connectHomepage` is a known, temporary niggle
* go to /products/vpn/resource-center/ which will redirect to /en-US/products/vpn/resource-center/
* go to change the locale in the URL to others and confirm they all end up on en-US  - eg /de/products/vpn/resource-center/ will redirect to en-US
* Browse to an article - you should see at least one complete article. If you see an empty card in your listing view, that's because you've pulled down a draft / WIP empty article -- check in Contentful to be sure.
* Confirm related article(s) are shown 
* Confirm homepage sync is still working as expected - try it by making edits to the sandboxed homepage

(We also have backend support for filtering by category, but this has been hidden in the frontend for now, until we know more about how we'll be using categories)
